src/steps/05-PerformanceRequirements/DOW/DOWLandingPage.vue

```diff
@@ -92,6 +116,123 @@ export default class DOWLandingPage extends Vue {
     }
   ];
 
+  get doesCurrentEnvExist(): boolean {
+    return CurrentEnvironment.currentEnvironment && 
+            CurrentEnvironment.currentEnvironment.current_environment_exists === "YES";
+  }
+
+  /**
+   * Sets the 'label' and 'isComplete' properties of the optimize/replicate section.
+   *
+   * Label gets built using the classification level options that the user selects. Default label
+   * is left as is, if no levels are selected
+   *
+   * Current functions are complete, if all the below are true;
+   * 1. If the user chooses NOT to optimize or replicate
+   * 2. If the user chooses to optimize or replicate AND
+   *      chooses (NO) or (YES for additional growth and enters anticipated capacity)
+   * 3. If the user chooses to optimize or replicate AND
+   *      chooses (NO) or (YES for phased approach and enters approach schedule)
+   */
+  setCurrentFunctionsProperties(): void {
+    const currEnv = CurrentEnvironment.currentEnvironment;
+    // is complete check
+    const reqSectionIndex =this.requirementSections
+      .findIndex(reqSection => reqSection.section === "ReplicateOptimize")
+    this.requirementSections[reqSectionIndex].isComplete =
+        (currEnv.current_environment_replicated_optimized === "NO") ||
+        ((currEnv.statement_replicated_optimized !== "") &&
+            ((currEnv.additional_growth === 'NO') || (currEnv.additional_growth === 'YES' &&
+                (currEnv.anticipated_yearly_additional_capacity as unknown as string) !== "")) &&
+            ((currEnv.has_phased_approach === "NO") ||
+                (currEnv.has_phased_approach === 'YES' && currEnv.phased_approach_schedule !== "")))
+    // label
+    let label = null;
+    if (currEnv.current_environment_replicated_optimized === "YES_REPLICATE") {
+      label = "Replicate (lift and shift) using JWCC offerings";
+    } else if (currEnv.current_environment_replicated_optimized === "YES_OPTIMIZE") {
+      label = "Optimize (improve/modernize) using JWCC offerings";
+    } else if (currEnv.current_environment_replicated_optimized === "NO") {
+      label = "No requirements";
+    }
+    if (label !== null) { // only set the label if the user had selected one of the options
+      this.requirementSections[reqSectionIndex].label = label;
+    }
+  }
+
+  /**
+   * Sets the 'label' and 'isComplete' properties of the architectural section.
+   *
```

Comment:
Initially selected "Yes" and completed the form for architectural design, and the card updated appropriately. Clicked "View/Edit" and changed it to "No", but card did not update to show "No Requirements" - If user selects "No", any previously entered arch design data should be cleared in store and SNOW.

Reply:
The data clearing and synching it up with SNOW issue also exists in Replicate/ Optimize section. Those errors were going to be fixed using separate bug/ task tickets. There should be no reason this should be treated differently. And once it is handled as part of that bug/ task, the bug described here will become irrelevant in the context of this story.